projects/frontend/cicd/.gitlab-ci.yml

```diff
@@ -9,6 +9,33 @@
   - "projects/frontend/cicd/**/*"
   - "projects/frontend/data-pipelines/**/*"
 
+frontend-data-pipelines-build:
+  image: "versatiledatakit/vdk-cicd-base-gui:2.0.0"
+  stage: build
+  needs: [frontend-shared-components-build]
+  before_script:
+    - cd projects/frontend/
+  script:
+    - ./cicd/install_data_pipelines.sh
+  coverage: /^TOTAL\s+\d+\s+\d+\s+(\d+\%)$/
+  retry: !reference [.retry, retry_options]
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "main" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
+      changes: *frontend_shared_components_locations
```

Comment:
The job needs `frontend-shared-components-build` for:
1. exporting `artifact: ..dist/..`
2. running also in case of `$CI_COMMIT_BRANCH == "main"`
we need to add on it 
cc @DeltaMichael

Reply:
Not sure what you mean by this. Maybe you can call the build_shared script as part of this job if that's what you're referring to.

Reply:
On release, we need to publish the same artifact build and already tested on build stage. Check out my changes, I'm resolving this  feel free to open if needed